src/modules/thirdParty/sagas.ts

```diff
@@ -110,4 +162,106 @@ export function* thirdPartySaga(builder: BuilderAPI) {
+      const { newItems, newItemCurations }: { newItems: Item[]; newItemCurations: ItemCuration[] } = yield call(
+        publishChangesToThirdPartyItems,
+        thirdParty,
+        items
+      )
+
+      yield put(publishThirdPartyItemsSuccess(collectionId, newItems, newItemCurations))
+      yield put(closeModal('PublishThirdPartyCollectionModal'))
+    } catch (error) {
+      yield put(publishThirdPartyItemsFailure(error.message))
+    }
+  }
+
+  function* pushChangesToThirdPartyItems(items: Item[]) {
+    const collectionId = getCollectionId(items)
+
+    const itemCurations: ItemCuration[] = yield select(getItemCurations, collectionId!)
+    const effects: CallEffect<ItemCuration>[] = items.map(item => {
+      const curation = itemCurations.find(itemCuration => itemCuration.itemId === item.id)
+      if (curation?.status === CurationStatus.PENDING) {
+        return call([builder, 'updateItemCurationStatus'], item.id, CurationStatus.PENDING)
+      }
+      return call([builder, 'pushItemCuration'], item.id) // FOR CURATIONS REJECTED/APPROVED
+    })
+
+    const newItemsCurations: ItemCuration[] = yield all(effects)
+    return newItemsCurations
+  }
+
+  function* handlePushChangesThirdPartyItemRequest(action: PushChangesThirdPartyItemsRequestAction) {
+    const { items } = action.payload
+    try {
+      const collectionId = getCollectionId(items)
+      const newItemsCurations: ItemCuration[] = yield call(pushChangesToThirdPartyItems, items)
+      yield put(pushChangesThirdPartyItemsSuccess(collectionId, newItemsCurations))
+      yield put(closeModal('PublishThirdPartyCollectionModal'))
+    } catch (error) {
+      yield put(pushChangesThirdPartyItemsFailure(error.message))
+    }
+  }
+
+  function* handlePublishAndPushChangesThirdPartyItemRequest(action: PublishAndPushChangesThirdPartyItemsRequestAction) {
+    const { thirdParty, itemsToPublish, itemsWithChanges } = action.payload
+    const collectionId = getCollectionId(itemsToPublish)
+    // We need to execute this two in secuence, because the push changes will create a new curation if there was one approved.
+    // It will create it with status PENDING, so the publish will fail if it's executed after that event.
+    // Publish items
+    try {
+      const resultFromPublish: { newItems: Item[]; newItemCurations: ItemCuration[] } = yield call(
+        publishChangesToThirdPartyItems,
+        thirdParty,
+        itemsToPublish
+      )
+      yield put(publishThirdPartyItemsSuccess(collectionId, resultFromPublish.newItems, resultFromPublish.newItemCurations))
+    } catch (error) {
+      yield put(publishThirdPartyItemsFailure(error.message))
+    }
+
+    // Push changes to items
+    try {
+      const resultFromPush: ItemCuration[] = yield call(pushChangesToThirdPartyItems, itemsWithChanges)
+      yield put(pushChangesThirdPartyItemsSuccess(collectionId, resultFromPush))
+    } catch (error) {
+      yield put(pushChangesThirdPartyItemsFailure(error.message))
```

Comment:
The two methods look good! but I think it's weird to have Success and Failure cases without their corresponding Requests, no?

Now that we have splitted the logic, we could just have a single Request (`PUBLISH_AND_PUSH_CHANGES_THIRD_PARTY_ITEMS_REQUEST`) and a single SUCCESS/FAILURE

```
PUBLISH_AND_PUSH_CHANGES_THIRD_PARTY_ITEMS_SUCCESS
PUBLISH_AND_PUSH_CHANGES_THIRD_PARTY_ITEMS_FAILURE
```

the failure should trigger if any method fails. The success if both work.

What do you think?

Reply:
I re-introduced `PUBLISH_AND_PUSH_CHANGES_THIRD_PARTY_ITEMS_SUCCESS` and `PUBLISH_AND_PUSH_CHANGES_THIRD_PARTY_ITEMS_FAILURE` 😂 Let me know what you think. Added some tests and the reducer logic as well.